dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -0,0 +1,179 @@
+describes how to make this highly available using replicated services.
+
+endif::[]
+
+
+[[using-a-router]]
+== Using a Router
+
+This is the most common way to access the cluster. A
+xref:../install_config/configuring_routing.adoc#install-config-configuring-routing[router]
+is configured to accept external requests and proxy them based on the
+configured routes.  This is limited to HTTP/HTTPS(SNI)/TLS(SNI) which
+covers web applications.
+
+An administrator can create a
+xref:../install_config/install/prerequisites.adoc#prereq-dns[wild-card DNS]
+entry, and then set up a router, after that the users can self-service
+the ingress without having to bother the admins. The router has controls
+to allow the admin to specify whether the users can self-provision
+hostnames, or if they must fit a pattern the admin defines. The other
+solutions require the admin to do it, or it requires that they delegate
+a lot of privilege.
+
+A set of routes can be created in the various projects. The overall
+set of routes is available to the set of routers. Each router selects
+from the set of routes.  All routers see all routes unless restricted
+by labels on the router, this is called router
+xref:../architecture/core_concepts/routes.adoc#router-sharding[sharding].
+
+The xref:../admin_guide/high_availability.adoc#admin-guide-high-availability[High Availability]
+section describes how to configure a router for high availability service
+using multiple replicas.
+
+[[using-the-loadbalancer]]
+== Using a Service of Type LoadBalancer
+
+There are load balancers available on
+xref:../install_config/configuring_aws.adoc#install-config-configuring-aws[AWS] and
+xref:../install_config/configuring_gce.adoc#install-config-configuring-gce[GCE] clouds as well as
+xref:../admin_guide/tcp_ingress_external_ports.adoc#admin-guide-expose-external-ports[non-cloud].
+
+The non-cloud
+xref:../admin_guide/tcp_ingress_external_ports.adoc#admin-guide-expose-external-ports[LoadBalancer] allocates
+a unique IP from a configured pool. This limits you to a single ingress IP
+(can be a VIP, but still will be a single machine for initial load balancing).
+It simplifies the admin job by providing the needed IP address, but uses one
+IP per service.
+
+Load balancing is described
+link:http://kubernetes.io/docs/user-guide/services/#type-loadbalancer[here].
+
+[[using-externalIP]]
+== Using a Service ExternalIP
+
+The user can manually assign
+link:http://kubernetes.io/docs/user-guide/services/#external-ips[ExternalIPs]
+to the service. The supplied list of IP addresses are used for load balancing
+incoming requests. The service port is opened on all nodes running kube-proxy.
+Doing this requires elevated permissions to assign, and manual tracking of the
+IP:ports that are in use.
+
+An externally visible IP can be configured in one of several ways:
+
+- manually configure the ExternalIP with a known external IP address.
+- configure ExternalIP to a
```

Comment:
VIPs are not generated from VRRP.  VRRP is a protocol to dynamically assign responsibility for a virtual router to a pool of hosts.  ``oadm ipfailover`` takes a ``--virtual-ips`` argument to specify which ip addresses to use.

A more accurate statement might be 'configure an ExternalIP corresponding to a virtual ip configured for high availability [link to high availability doc]'.